src/main/java/com/bandwidth/iris/sdk/model/AvailableNumbers.java

```diff
@@ -1,27 +0,0 @@
-package com.bandwidth.iris.sdk.model;
-
-import com.bandwidth.iris.sdk.IrisClient;
-import com.bandwidth.iris.sdk.IrisConstants;
-import com.bandwidth.iris.sdk.IrisResponse;
-import com.bandwidth.iris.sdk.utils.XmlUtils;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-
-/**
- * Created by sbarstow on 11/19/14.
- */
-public class AvailableNumbers {
-    public static List<?> search(IrisClient client, Map<String, Object> query) throws Exception {
```

Comment:
I'd prefer two "search" methods. One for "basic" and one for "detail". That will avoid the rather unfortunate `returnDetails` code and it will avoid returning two types of lists. `TelephoneNumber` and `TelephoneNumberDetail`. The generic `List<?>` with a wildcard could be avoided.